Makefile

```diff
@@ -390,14 +371,8 @@ generate-k8s-api: ## Generate Cilium k8s API client, deepcopy and deepequal Go s
 	$(eval DEEPEQUAL_PACKAGES := $(shell grep "\+deepequal-gen" -l -r --include \*.go --exclude-dir 'vendor' . | xargs dirname {} | sort | uniq | grep -x -v '.' | sed 's|\.\/|github.com/cilium/cilium\/|g'))
 	$(QUIET) $(call generate_deepequal,${DEEPEQUAL_PACKAGES},"$(TMPDIR)")
 
-	$(eval DEEPCOPY_PACKAGES := $(shell grep "\+k8s:deepcopy-gen" -l -r --include \*.go --exclude-dir 'vendor' . | xargs dirname {} | sort | uniq | grep -x -v '.' | sed 's|\.\/|github.com/cilium/cilium\/|g'))
-	$(QUIET) $(call generate_deepcopy,${DEEPCOPY_PACKAGES},"$(TMPDIR)")
-
-	$(QUIET) $(call generate_k8s_api,client,github.com/cilium/cilium/pkg/k8s/slim/k8s/client,github.com/cilium/cilium/pkg/k8s/slim/k8s/api,"discovery:v1beta1 discovery:v1 networking:v1 core:v1","$(TMPDIR)")
-	$(QUIET) $(call generate_k8s_api,client,github.com/cilium/cilium/pkg/k8s/slim/k8s/apiextensions-client,github.com/cilium/cilium/pkg/k8s/slim/k8s/apis,"apiextensions:v1","$(TMPDIR)")
-	$(QUIET) $(call generate_k8s_api,client$(comma)lister$(comma)informer,github.com/cilium/cilium/pkg/k8s/client,github.com/cilium/cilium/pkg/k8s/apis,$(GEN_CRD_GROUPS),"$(TMPDIR)")
+	$(QUIET) contrib/scripts/k8s-code-gen.sh "$(TMPDIR)"
 
-	$(QUIET) cp -r "$(TMPDIR)/github.com/cilium/cilium/." ./
```

Comment:
nit: Given that there are various steps inside this Make target which generate files within the TMPDIR, I don't think it makes sense to hide the copy inside the `k8s-code-gen.sh` script.

Reply:
This part was added by @aanm, Andre is there a reason for moving the `cp` into the script?

Reply:
@joestringer @christarazi yes, the `cp` needs to be done before the code generation for the remaining of the packages. You can test it out by removing the `cp` from the script and re-add it in the Makefile to see the difference of the results

Reply:
I see. Probably the solution to my nit is to move even more of the build process out of make, but that'd be a bigger ask. Feels messy to me, but nothing to hold up this PR over.